internal/impl/azure/output_cosmosdb.go

```diff
@@ -0,0 +1,151 @@
+package azure
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+
+	"github.com/benthosdev/benthos/v4/internal/component/output"
+	"github.com/benthosdev/benthos/v4/internal/impl/azure/cosmosdb"
+	"github.com/benthosdev/benthos/v4/public/service"
+)
+
+const (
+	cdboFieldBatching = "batching"
+)
+
+func cosmosDBOutputConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		// Stable(). TODO
+		Categories("Azure").
+		Version("v4.25.0").
+		Summary("Creates or updates messages as JSON documents in [Azure CosmosDB](https://learn.microsoft.com/en-us/azure/cosmos-db/introduction).").
+		Description(output.Description(true, true, `
+When creating documents, each message must have the `+"`id`"+` property (case-sensitive) set (or use `+"`auto_id: true`"+`). It is the unique name that identifies the document, that is, no two documents share the same `+"`id`"+` within a logical partition. The `+"`id`"+` field must not exceed 255 characters. More details can be found [here](https://learn.microsoft.com/en-us/rest/api/cosmos-db/documents).
+
+The `+"`partition_keys`"+` field must resolve to the same value(s) across the entire message batch.
+`+cosmosdb.CredentialsDocs+cosmosdb.BatchingDocs)).
+		Footnotes(cosmosdb.EmulatorDocs).
+		Fields(cosmosdb.ContainerClientConfigFields()...).
+		Field(cosmosdb.PartitionKeysField(true)).
+		Fields(cosmosdb.CRUDFields(false)...).
+		Field(service.NewBatchPolicyField(cdboFieldBatching)).
+		Field(service.NewOutputMaxInFlightField()).
+		LintRule("root = []"+cosmosdb.CommonLintRules+cosmosdb.CRUDLintRules).
```

Comment:
This is a tiny bit awkward, so I'm kinda' wondering if maybe it's worth having a `LintRules()` API which does `root = []` implicitly and takes as input a variadic list of strings, each containing some bloblang snippet with one or more `root."-" = ...` and maybe extra variables. I don't really like it, though, since it would be hard to spot bugs in the composed mapping.